daskernetes/core.py

```diff
@@ -165,24 +212,104 @@ def _make_pod(self):
             )
         )
 
+
+        resources = client.V1ResourceRequirements(limits={}, requests={})
+
+        if self.cpu_request:
+            resources.requests['cpu'] = self.cpu_request
+        if self.memory_request:
+            resources.requests['memory'] = self.memory_request
+
+        if self.cpu_limit:
+            resources.limits['cpu'] = self.cpu_limit
+        if self.memory_limit:
+            resources.limits['memory'] = self.memory_limit
+
+        pod.spec.containers[0].resources = resources
+
+        for key, value in self.extra_container_config.items():
+            self._set_k8s_attribute(
+                pod.spec.containers[0],
+                key,
+                value
+            )
+
+        for key, value in self.extra_pod_config.items():
+            self._set_k8s_attribute(
+                pod.spec,
+                key,
+                value
+            )
+        return pod
+
+    def _set_k8s_attribute(self, obj, attribute, value):
```

Comment:
It looks like the only reference to self is to get out core_api.  Perhaps this should be a standalone function that accepts an api object?  That might also make it easier to test.

Reply:
good idea, I'll do that!

I think the serialize function there could be a static or classmethod, I'll file a bug upstream.